src/CitMovie.Data/Repositories/PersonRepository.cs

```diff
@@ -17,43 +17,136 @@ public async Task<IEnumerable<Person>> GetPersonsAsync(int page, int pageSize)
             .Pagination(page, pageSize)
             .ToListAsync();
     }
-    public async Task<int> GetTotalPersonsCountAsync()
-    {
-        return await _dataContext.People.CountAsync();
+
+    public async Task<IEnumerable<Person>> GetPersonsByNameAsync(string name, int? userId, int page, int pageSize) {
+        IQueryable<int> ids = _dataContext
+            .PersonSearch(name, userId)

```

Comment:
Now when we are using our own SQL function "structured_string_search_name" are we then sure that it is safe against SQL-Injection? Here in this function we are just passing the raw data to the DB function, so we have no protection here in the backend.

Reply:
Entitiy framework uses parameters for all generated queries. See example on picture

![image](https://github.com/user-attachments/assets/e06d4202-ed06-4460-a00d-e3a877fc512a)